### PR TITLE
SSUI Designer: New Node Toolbars and Connection Mode

### PR DESCRIPTION
**Pull request by dtaylor113**
Added toolbar under each item/node on canvas:

image

Implemented new Connection Mode:

image

image

image

Added Generic Items:

image

### CODE REVIEW

**Comment by dtaylor113** [comment]
@serenamarie125  "ping" :-)

**Comment by chriskacerguis** [comment]
👍

File: client/app/components/flowchart/flowchart_template.html [hunk 1]
```diff
@@ -1,29 +1,33 @@
...
-      <feGaussianBlur result="blurOut" in="matrixOut" stdDeviation="6" />
-      <feBlend in="SourceGraphic" in2="blurOut" mode="normal" />
-    </filter>
-  </defs>
 
+  <!-- Connection Mode Notification -->
+  <g ng-if="inConnectingMode">
+    <rect class="connecting-mode-rec" ry="1" rx="1" x="0" y="0" width="640" height="32" />
+    <text class="connecting-mode-label" x="12" y="22" ng-if="availableConnections()">
+      Select a second item to complete the connection or click on the canvas to cancel
```
Comment (mzazrivec) [inline client/app/components/flowchart/flowchart_template.html]: This string needs to be translated.

File: client/app/components/flowchart/flowchart_template.html [hunk 2]
```diff
@@ -1,29 +1,33 @@
...
+    </text>
+    <text class="connecting-mode-label-warning" x="12" y="22" ng-if="!availableConnections()">
+      No available connections! Click on the canvas to cancel
```
Comment (mzazrivec) [inline client/app/components/flowchart/flowchart_template.html]: This string needs to be translated.

Comment (dtaylor113) [inline client/app/components/flowchart/flowchart_template.html]: Hi, @mzazrivec , these are fixed and will be included in my next PR. -thanks

**Pull request merged**